studio/app/optinist/core/edit_ROI/wrappers/suite2p_edit_roi/commit.py

```diff
@@ -0,0 +1,103 @@
+    EditRoiData,
+    FluoData,
+    IscellData,
+    RoiData,
+    Suite2pData,
+)
+from studio.app.optinist.schemas.roi import RoiPos
+
+
+def commit_edit(data: EditRoiData, ops: Suite2pData, iscell, node_dirpath, function_id):
+    from suite2p.detection.stats import median_pix, roi_stats
+
+    from studio.app.optinist.core.edit_ROI.edit_ROI import CellType
+
+    ops = ops.data
+    stat = ops["stat"]
+    temp_roi_data_dict = {**data.temp_add_roi, **data.temp_merge_roi}
+    temp_roi_data_dict = dict(sorted(temp_roi_data_dict.items()))
+
+    if "aspect" in ops:
+        d0 = np.array([int(ops["aspect"] * 10), 10])
+    else:
+        d0 = ops["diameter"]
+        if isinstance(d0, int):
+            d0 = [d0, d0]
+
+    for roi_id, roi_data in temp_roi_data_dict.items():
+        iscell[int(roi_id)] = CellType.ROI
+        # Save fluorescence traces of added roi
+        if isinstance(roi_data, RoiPos):
+            stat0 = get_stat0_add_roi(
+                ops, roi_data.posx, roi_data.posy, roi_data.sizex, roi_data.sizey
+            )
+            Fcell, Fneu, _, _, _, ops, _ = masks_and_traces(ops, stat0, stat)
+            ops["F"] = np.concatenate((ops["F"], Fcell), axis=0)
+            ops["Fneu"] = np.concatenate((ops["Fneu"], Fneu), axis=0)
+
+        # Save fluorescence traces of merged roi
+        elif isinstance(roi_data, list):
+            ypix = np.zeros((0,), np.int32)
+            xpix = np.zeros((0,), np.int32)
+            lam = np.zeros((0,), np.float32)
+
+            merged_cells = np.unique(np.array(roi_data))
+            F = np.mean((ops["F"][roi_data, :]), axis=0)
+            Fneu = np.mean((ops["Fneu"][roi_data, :]), axis=0)
+
+            # merge cells regions
+            for n in merged_cells:
+                ypix = np.append(ypix, stat[n]["ypix"])
+                xpix = np.append(xpix, stat[n]["xpix"])
+                lam = np.append(lam, stat[n]["lam"])
+
+            # remove overlaps from merged cells regions
+            ipix = np.concatenate((ypix[:, np.newaxis], xpix[:, np.newaxis]), axis=1)
+            _, goodi = np.unique(ipix, return_index=True, axis=0)
+            ypix = ypix[goodi]
+            xpix = xpix[goodi]
+            lam = lam[goodi]
+
+            # compute statistics of merges
+            stat0 = {}
+            stat0["ypix"] = ypix
+            stat0["xpix"] = xpix
+            stat0["lam"] = lam / lam.sum()
+            stat0["npix"] = ypix.size
+            stat0["med"] = median_pix(ypix, xpix)
+            stat0["chan2_prob"] = -1
+            stat0["inmerge"] = -1
+
+            ops["F"] = np.vstack((ops["F"], F))
+            ops["Fneu"] = np.vstack((ops["Fneu"], Fneu))
+
+            stat.append(stat0)
+            stat = roi_stats(stat, d0[0], d0[1], ops["Ly"], ops["Lx"])
+            stat[-1]["lam"] = stat[-1]["lam"] * merged_cells.size
+
+    iscell[iscell == CellType.TEMP_DELETE] = CellType.NON_ROI
+
+    ops["stat"] = stat
+    data.commit()
```

Comment:
@quanpython 
We have added an assumption of the BE data retention structure until the ROI is committed to the following document.

- https://docs.google.com/presentation/d/1Pz7N_bOCs2SuHq9RHHxZZHd79fnXhEHt/
  - slide 2. ROI EDIT data oeperation sequence

Please confirm the contents and apply adjustments.

- Issues in the current implementation
  - Overwriting existing log data (pkl, json, etc.) by ROI editing is not assumed. (Prepare a separate file for cache).
  - State management of the cache is a little unclear. (When the screen is redisplayed, the data in the process of editing in the past may be redisplayed. etc.)

Reply:
@quanpython 
Sorry for the late reply.

Initialize is assumed to be [at the start of ROI Edit (click on the ROI Edit link)].

Reply:
@itutu-tienday
To meet the above requirement, I think the FE side will need to call the cancel API at the time of clicking the link, is it okay?

Reply:
@quanpython 
I would appreciate suggestions as to whether a cancel API would be appropriate (or is an edit start API appropriate?). I think some request from FE to BE is necessary to clear the previous cache data at the start of editing.
Please let me know if you have any concerns.

Reply:
@itutu-tienday
Temp data still displays after browser reload until clicking on the ROI Edit link, is it reasonable?

Reply:
@quanpython 
Temporary data is also deleted by clicking on the Cancel link. Is there a difference in understanding on this point?

Also, where do you expect the browser reload to take place (before, after, or both, etc.)? (Before, after, or both of ROI Edit, etc.)

Reply:
@itutu-tienday

In case the browser reloads while editing, I go to Visualize -> open newplot -> click Select ROI -> cell_roi, the temporary data is still displayed because the Cancel API is not called until the ROI Edit link is clicked.

Do you want to call the Cancel API at the time of selecting `cell_roi` from the `Select ROI` dropdown instead of clicking the Edit ROI link?

Reply:
@quanpython 
I see.
I think it would be appropriate to not display cached data (display the last committed data) before the start of ROI Edit, but is the implementation of this behavior complicated?

Reply:
@itutu-tienday
Because BE can not control it, I will contact FE side about this issue.